src/luascript.cpp

```diff
@@ -10824,6 +10830,45 @@ int LuaScriptInterface::luaPlayerGetIdleTime(lua_State* L)
 	return 1;
 }
 
+// OfflinePlayer
+int LuaScriptInterface::luaOfflinePlayerCreate(lua_State* L)
+{
+	// OfflinePlayer(id or name)
+	std::unique_ptr<Player> player_ptr = std::make_unique<Player>(nullptr);
+	if (isNumber(L, 2)) {
+		const uint32_t playerId = getNumber<uint32_t>(L, 2);
+		if (!IOLoginData::loadPlayerById(player_ptr.get(), playerId)) {
+			lua_pushnil(L);
+			return 1;
+		}
+	} else if (isString(L, 2)) {
+		const std::string& playerName = getString(L, 2);
+		if (!IOLoginData::loadPlayerByName(player_ptr.get(), playerName)) {
+			lua_pushnil(L);
+			return 1;
+		}
+	}
+
+	pushUserdata<Player>(L, player_ptr.release());
+	setMetatable(L, -1, "OfflinePlayer");
+	return 1;
+}
+
+int LuaScriptInterface::luaOfflinePlayerDelete(lua_State* L)
```

Comment:
`delete` instead `free`, just for consistency with TFS nomenclature.

In my personal opinion I would use `remove` this way I override the `remove` method of Creatures, since this method causes interpreter crashes with this object OfflinePlayer, so no one would use it accidentally.

Example:
```c++
registerMetaMethod("OfflinePlayer", "__gc", LuaScriptInterface::luaOfflinePlayerRemove);

registerMethod("OfflinePlayer", "remove", LuaScriptInterface::luaOfflinePlayerRemove);

int LuaScriptInterface::luaOfflinePlayerRemove(lua_State* L)
{
	// offlinePlayer:remove() offlinePlayer:__gc()
	Player** offlinePlayerPtr = getRawUserdata<Player>(L, 1);
	if (auto offlinePlayer = *offlinePlayerPtr) {
		g_game.ReleaseCreature(offlinePlayer);
		*offlinePlayerPtr = nullptr;
		pushBoolean(L, true);
		return 1;
	}

	lua_pushnil(L);
	return 1;
}
```

If this object is based on the Player/Creaturewe should treat the object as such, so use `g_game.ReleaseCreature`

However, I'm not from the development team so don't take my suggestions too seriously and my personal opinions 😊.

Reply:
ReleaseCreature is not used for creatures that are loaded like that, they are not stored anywhere so there is no need to delay the deletion. Delete operator is ok